wgpu-types/src/lib.rs

```diff
@@ -83,11 +83,21 @@ impl From<Backend> for BackendBit {
     }
 }
 
+/// Extensions requested or supported by a device.
+///
+/// Before an extension is used, one must check the actual supported
+/// extensions by ...
+///
+/// TODO: Allow user to check extensions available.
 #[repr(C)]
 #[derive(Clone, Debug, Default, PartialEq, Eq, Hash)]
 #[cfg_attr(feature = "trace", derive(Serialize))]
 #[cfg_attr(feature = "replay", derive(Deserialize))]
 pub struct Extensions {
```

Comment:
Should Extensions/Limits be marked as `#[non_exhaustive]`?

Reply:
I would imagine so. I should add that note to #691. I anticipate discussion on that issue changing this PR before it's merged.

Reply:
I realize now that another benefit to this is that adding extensions that don't modify structures become a non breaking change.

Reply:
Actually, we may need to add extensions that *do* change structures in a non-breaking way, at some point, probably soon. I.e. we don't want to break wgpu-core API after WebGPU is out and we are just adding more extensions :/

We could do this by adding a private `PhantomData` field to structures, which implement `Default`.

Reply:
If we're okay with the MSRV bump, we can just use `#[non_exhaustive]` on basically every struct.